Classes/Domain/NodeCreation/NodeMutator.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flowpack\NodeTemplates\Domain\NodeCreation;
+
+use Neos\ContentRepository\Domain\Model\NodeInterface;
+use Neos\ContentRepository\Domain\NodeAggregate\NodeName;
+use Neos\ContentRepository\Domain\NodeType\NodeTypeName;
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Core\Bootstrap;
+use Neos\Neos\Service\NodeOperations;
+
+/**
+ * We queue changes on nodes instead of making them directly.
+ * This allows us to check if the template is valid by cr perspective, without actually applying it,
+ * as all checks are executed before the Mutators are run.
+ *
+ * A mutator should only queue a mutation, that is guaranteed to succeed, as there is no exception handing.
+ *
+ * @Flow\Proxy(false)
+ */
+class NodeMutator
+{
+    private \Closure $mutator;
+
+    /**
+     * @param \Closure(NodeInterface $currentNode): ?NodeInterface $mutator
+     */
+    private function __construct(
+        \Closure $mutator
+    ) {
+        $this->mutator = $mutator;
+    }
+
+    /**
+     * Queues to execute this mutator on the current node
+     *
+     * @param \Closure(NodeInterface $currentNode): ?NodeInterface $mutator
+     */
+    public static function unsafeFromClosure(\Closure $mutator): self
+    {
+        return new self($mutator);
+    }
+
+    /**
+     * Queues to execute the {@see NodeMutatorCollection} on the current node but the operations wont change the current node.
```

Comment:
This comment doesn't make sense to me, shouldn't it say it won't change the parent node or something?
Also the word `collection` is missing after the `the` I think 🤔